Lib/importlib/_bootstrap_external.py

```diff
@@ -326,10 +347,13 @@ def source_from_cache(path):
         raise NotImplementedError('sys.implementation.cache_tag is None')
     path = _os.fspath(path)
     head, pycache_filename = _path_split(path)
-    head, pycache = _path_split(head)
-    if pycache != _PYCACHE:
-        raise ValueError('{} not bottom-level directory in '
-                         '{!r}'.format(_PYCACHE, path))
+    if bytecode_path is not None and head.startswith(bytecode_path + path_sep):
+        head = head[len(bytecode_path):]
+    else:
+        head, pycache = _path_split(head)
+        if pycache != _PYCACHE:
+            raise ValueError('{} not bottom-level directory in '
```

Comment:
f-string?

Reply:
I think Eric meant "f-strings everywhere!" 😉

Reply:
Sure, can do, I left it unchanged because this is pre-existing code, all I did was indent it one level. I'm all in favor of f-strings everywhere :-)